src/pages/Lending/Pool/components/Repay/RepayConfirm.tsx

```diff
@@ -1,34 +1,230 @@
-  const debtAsset = useAppSelector(state => selectAssetById(state, usdcAssetId))
+  const collateralAsset = useAppSelector(state => selectAssetById(state, collateralAssetId))
+
   const handleBack = useCallback(() => {
     history.push(RepayRoutePaths.Input)
   }, [history])
+
   const divider = useMemo(() => <Divider />, [])
 
-  if (!collateralAsset || !debtAsset) return null
+  const { data: lendingPositionData } = useLendingPositionData({
+    assetId: collateralAssetId,
+    accountId: collateralAccountId,
+  })
+
+  const userCurrencyToUsdRate = useAppSelector(selectUserCurrencyToUsdRate)
+
+  const debtBalanceUserCurrency = useMemo(() => {
+    return bnOrZero(lendingPositionData?.debtBalanceFiatUSD ?? 0)
+      .times(userCurrencyToUsdRate)
+      .toFixed()
+  }, [lendingPositionData, userCurrencyToUsdRate])
+
+  const repaymentAmountFiatUserCurrency = useMemo(() => {
+    if (!lendingPositionData) return null
+
+    const proratedCollateralFiatUserCurrency = bnOrZero(repaymentPercent)
+      .times(debtBalanceUserCurrency)
+      .div(100)
+
+    return proratedCollateralFiatUserCurrency.toFixed()
+  }, [debtBalanceUserCurrency, lendingPositionData, repaymentPercent])
+
+  const repaymentAssetMarketData = useAppSelector(state =>
+    selectMarketDataById(state, repaymentAsset?.assetId ?? ''),
+  )
+
+  const repaymentAmountCryptoPrecision = useMemo(() => {
+    if (!repaymentAmountFiatUserCurrency) return null
+
+    return bnOrZero(repaymentAmountFiatUserCurrency).div(repaymentAssetMarketData.price).toFixed()
+  }, [repaymentAmountFiatUserCurrency, repaymentAssetMarketData.price])
+
+  const useLendingQuoteCloseQueryArgs = useMemo(
+    () => ({
+      collateralAssetId,
+      repaymentAssetId: repaymentAsset?.assetId ?? '',
+      repaymentPercent,
+      repaymentAccountId,
+      collateralAccountId,
+    }),
+    [
+      collateralAccountId,
+      collateralAssetId,
+      repaymentAccountId,
+      repaymentAsset?.assetId,
+      repaymentPercent,
+    ],
+  )
+
+  const { data: lendingQuoteCloseData, isLoading: isLendingQuoteCloseLoading } =
+    useLendingQuoteCloseQuery(useLendingQuoteCloseQueryArgs)
+
+  const chainAdapter = getChainAdapterManager().get(
+    fromAssetId(repaymentAsset?.assetId ?? '').chainId,
+  )
+  const selectedCurrency = useAppSelector(selectSelectedCurrency)
+
+  // TODO(gomes): handle error (including trading halted) and loading states here
+  const handleRepay = useCallback(async () => {
+    if (
+      !(
+        repaymentAsset &&
+        wallet &&
+        supportsETH(wallet) &&
+        chainAdapter &&
+        lendingQuoteCloseData &&
+        repaymentAmountCryptoPrecision
+      )
+    )
+      return
+
+    const supportedEvmChainIds = getSupportedEvmChainIds()
+    const estimatedFees = await estimateFees({
+      cryptoAmount: repaymentAmountCryptoPrecision,
+      assetId: repaymentAsset.assetId,
+      memo: supportedEvmChainIds.includes(fromAssetId(repaymentAsset.assetId).chainId)
```

Comment:
What's the reason behind only hexlifying the quote memo if the asset's chain is a supported EVM chain?

Reply:
chain-adapters (and hdwallet/ethers down the stack) expects a hex memo for EVM chains, but other chains are a regular UTF-8 one!